ultramodern/src/renderer_context.cpp

```diff
@@ -0,0 +1,70 @@
+#include <cassert>
+#include <mutex>
+
+#include "ultramodern/renderer_context.hpp"
+#include "ultramodern/ultramodern.hpp"
+
+static ultramodern::renderer::callbacks_t render_callbacks{};
+
+void ultramodern::renderer::set_callbacks(const callbacks_t& callbacks) {
+    render_callbacks = callbacks;
+}
+
+
+std::unique_ptr<ultramodern::renderer::RendererContext> ultramodern::renderer::create_render_context(uint8_t* rdram, WindowHandle window_handle, bool developer_mode) {
+    if (render_callbacks.create_render_context == nullptr) {
+        error_handling::message_box("[Error] The mandatory render callback `create_render_context` was not registered");
+        // TODO: should we make a macro for this?
```

Comment:
Seems like it depends on the standard library version rather than the compiler itself.
![image](https://github.com/N64Recomp/N64ModernRuntime/assets/7416381/ed3d5429-4d47-4547-8d64-ce5a75757290)
https://en.cppreference.com/w/cpp/compiler_support

I'll give it a try and see how it goes

Reply:
I just realized that I'm not sure how to check for the release dates of those libraries, neither how to check which one I have installed.
I guess I'll go back to a plain macro. A bit of a shame, `std::source_location` is a very clean solution